bigquery_etl/metadata/validate_metadata.py

```diff
@@ -27,6 +32,65 @@ def validate_public_data(metadata, path):
     return is_valid
 
 
+def validate_change_control(
+    file_path,
+    metadata,
+    codeowners_file,
+    project_id="moz-fx-data-shared-prod",
+    sql_dir="sql",
+):
+    """Verify that a query is correctly setup for change control."""
+    path_to_add = file_path.partition(f"{project_id}/")[2]
+    path_in_codeowners = os.path.join(sql_dir, project_id, path_to_add)
+    has_change_control = CHANGE_CONTROL_LABEL in metadata.labels
+
+    if has_change_control:
+        # This label requires to have at least one owner in the metadata file.
+        # And for any of the owners, at least one entry in the CODEOWNERS file.
+        # The owners can be emails or GitHub identities e.g. mozilla/team_name.
+        if len(metadata.owners) == 0:
+            click.echo(
+                click.style(
+                    f"The metadata for {file_path} has the label"
+                    f" change_controlled but it's missing owners."
+                )
+            )
+            return
+
+        with open(codeowners_file, "r") as owners_file:
+            content = owners_file.readlines()
+        owners_list = []
+        for owner in metadata.owners:
+            if "@" not in owner:
+                owner = f"@{owner}"
+            owners_list.append(owner)
+        sample_row_all_owners = f"/{path_in_codeowners} {(' '.join(owners_list))}"
+
+        if not [line for line in content if path_in_codeowners in line]:
```

Comment:
Since the logic has been refactored back to using substring comparisons, I think the comments from my first review are relevant again about accepting commented strings e.g. I would expect this to erroneously pass the check:

```
# {path_in_codeowners} {valid_owner}
```

Can you add a test for this, as well as a test verifying that from my second review a line like:

```
{path_in_codeowners} {valid_owner_1} {valid_owner_2}
```

is now accepted?

Reply:
Agree, added a few test scenarios for github identities, one/none/many owners in CODEOWNERS and the handling of lines starting with '#"